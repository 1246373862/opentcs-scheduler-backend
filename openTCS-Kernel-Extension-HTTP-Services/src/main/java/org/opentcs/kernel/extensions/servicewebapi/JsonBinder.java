/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import java.io.IOException;

/**
 * Binds JSON strings to objects and vice versa.
 */
public class JsonBinder {

  /**
   * Maps between objects and their JSON representations.
   */
  private final ObjectMapper objectMapper
      = new ObjectMapper()
          .registerModule(new JavaTimeModule())
          .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);

  /**
   * Creates a new instance.
   */
  public JsonBinder() {
  }

  /**
   * Maps the given JSON string to an object.
   *
   * @param <T> The type of object to map to.
   * @param jsonString The JSON string.
   * @param clazz The type of object to map to.
   * @return The object created from the JSON string.
   * @throws IllegalArgumentException In case there was a problem mapping the given object from
   * JSON.
   * (An IllegalArgumentException is mapped to HTTP status code 400, indicating a client error.)
   */
  public <T> T fromJson(String jsonString, Class<T> clazz)
      throws IllegalArgumentException {
    try {
      return objectMapper.readValue(jsonString, clazz);
    }
    catch (IOException exc) {
      throw new IllegalArgumentException("Could not parse JSON input", exc);
    }
  }

  /**
   * Maps the given object to a JSON string.
   *
   * @param object The object to be mapped.
   * @return The JSON string representation of the object.
   * @throws IllegalStateException In case there was a problem mapping the given object to JSON.
   * (An IllegalStateException is mapped to HTTP status code 500, indicating an internal error.)
   */
  public String toJson(Object object)
      throws IllegalStateException {
    try {
      return objectMapper
          .writerWithDefaultPrettyPrinter()
          .writeValueAsString(object);
    }
    catch (JsonProcessingException exc) {
      throw new IllegalStateException("Could not produce JSON output", exc);
    }
  }

  /**
   * Maps the given throwable to a JSON string.
   *
   * @param t The throwable to be mapped.
   * @return A JSON string for the given throwable, consisting of a single-element array containing
   * the throwable's message.
   * @throws IllegalStateException In case there was a problem mapping the given object to JSON.
   * (An IllegalStateException is mapped to HTTP status code 500, indicating an internal error.)
   */
  public String toJson(Throwable t)
      throws IllegalStateException {
    try {
      return objectMapper
          .writerWithDefaultPrettyPrinter()
          .writeValueAsString(objectMapper.createArrayNode().add(t.getMessage()));
    }
    catch (JsonProcessingException exc) {
      throw new IllegalStateException("Could not produce JSON output", exc);
    }
  }

}
