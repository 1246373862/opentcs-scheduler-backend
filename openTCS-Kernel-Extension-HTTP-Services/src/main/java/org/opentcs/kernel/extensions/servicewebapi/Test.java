package org.opentcs.kernel.extensions.servicewebapi;

import org.opentcs.components.kernel.KernelExtension;

public class Test implements KernelExtension {

  private volatile boolean enabled;

  @Override
  public void initialize() {
    if (enabled){
      return;
    }
    System.out.println("初始化~~~~~初始化~~~~~初始化~~~~~初始化~~~~~初始化~~~~~初始化~~~~~");
  }

  @Override
  public boolean isInitialized() {
    return enabled;
  }

  @Override
  public void terminate() {
    if (!enabled) {
      return;
    }
    enabled = false;
    System.out.println("结束！！");
  }
}
