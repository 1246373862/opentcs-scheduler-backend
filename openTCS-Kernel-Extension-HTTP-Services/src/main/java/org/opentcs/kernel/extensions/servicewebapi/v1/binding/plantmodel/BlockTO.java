/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel;

import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.PropertyTO;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import static java.util.Objects.requireNonNull;
import java.util.Set;
import javax.annotation.Nonnull;
import org.opentcs.data.model.Block;

/**
 */
public class BlockTO {

  private String name;
  private String type = Block.Type.SINGLE_VEHICLE_ONLY.name();
  private Layout layout = new Layout();
  private Set<String> memberNames = Set.of();
  private List<PropertyTO> properties = List.of();

  @JsonCreator
  public BlockTO(@Nonnull @JsonProperty(value = "name", required = true) String name) {
    this.name = requireNonNull(name, "name");
  }

  @Nonnull
  public String getName() {
    return name;
  }

  public BlockTO setName(@Nonnull String name) {
    this.name = requireNonNull(name, "name");
    return this;
  }

  @Nonnull
  public List<PropertyTO> getProperties() {
    return properties;
  }

  public BlockTO setProperties(@Nonnull List<PropertyTO> properties) {
    this.properties = requireNonNull(properties, "properties");
    return this;
  }

  @Nonnull
  public String getType() {
    return type;
  }

  public BlockTO setType(@Nonnull String type) {
    this.type = requireNonNull(type, "type");
    return this;
  }

  @Nonnull
  public Layout getLayout() {
    return layout;
  }

  public BlockTO setLayout(@Nonnull Layout layout) {
    this.layout = requireNonNull(layout, "layout");
    return this;
  }

  @Nonnull
  public Set<String> getMemberNames() {
    return memberNames;
  }

  public BlockTO setMemberNames(@Nonnull Set<String> memberNames) {
    this.memberNames = requireNonNull(memberNames, "memberNames");
    return this;
  }

  public static class Layout {

    private String color = "#FF0000";

    public Layout() {
    }

    @Nonnull
    public String getColor() {
      return color;
    }

    public Layout setColor(@Nonnull String color) {
      this.color = requireNonNull(color, "color");
      return this;
    }

  }

}
