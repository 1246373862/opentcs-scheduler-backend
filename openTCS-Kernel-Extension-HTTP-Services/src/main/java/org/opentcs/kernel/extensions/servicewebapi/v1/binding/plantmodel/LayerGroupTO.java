/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import static java.util.Objects.requireNonNull;
import javax.annotation.Nonnull;

/**
 */
public class LayerGroupTO {

  private int id;
  private String name;
  private boolean visible;

  @JsonCreator
  public LayerGroupTO(@JsonProperty(value = "id", required = true) int id,
                      @Nonnull @JsonProperty(value = "name", required = true) String name,
                      @JsonProperty(value = "visible", required = true) boolean visible) {
    this.id = id;
    this.name = requireNonNull(name, "name");
    this.visible = visible;
  }

  public int getId() {
    return id;
  }

  public LayerGroupTO setId(int id) {
    this.id = id;
    return this;
  }

  @Nonnull
  public String getName() {
    return name;
  }

  public LayerGroupTO setName(@Nonnull String name) {
    this.name = requireNonNull(name, "name");
    return this;
  }

  public boolean isVisible() {
    return visible;
  }

  public LayerGroupTO setVisible(boolean visible) {
    this.visible = visible;
    return this;
  }

}
