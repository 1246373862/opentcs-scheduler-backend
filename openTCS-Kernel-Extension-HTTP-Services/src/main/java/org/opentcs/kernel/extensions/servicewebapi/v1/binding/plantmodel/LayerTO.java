/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import static java.util.Objects.requireNonNull;
import javax.annotation.Nonnull;

/**
 */
public class LayerTO {

  private int id;
  private int ordinal;
  private boolean visible;
  private String name;
  private int groupId;

  @JsonCreator
  public LayerTO(@JsonProperty(value = "id", required = true) int id,
                 @JsonProperty(value = "ordinal", required = true) int ordinal,
                 @JsonProperty(value = "visible", required = true) boolean visible,
                 @Nonnull @JsonProperty(value = "name", required = true) String name,
                 @JsonProperty(value = "groupId", required = true) int groupId) {
    this.id = id;
    this.ordinal = ordinal;
    this.visible = visible;
    this.name = requireNonNull(name, "name");
    this.groupId = groupId;
  }

  public int getId() {
    return id;
  }

  public LayerTO setId(int id) {
    this.id = id;
    return this;
  }

  public int getOrdinal() {
    return ordinal;
  }

  public LayerTO setOrdinal(int ordinal) {
    this.ordinal = ordinal;
    return this;
  }

  public boolean isVisible() {
    return visible;
  }

  public LayerTO setVisible(boolean visible) {
    this.visible = visible;
    return this;
  }

  @Nonnull
  public String getName() {
    return name;
  }

  public LayerTO setName(@Nonnull String name) {
    this.name = requireNonNull(name, "name");
    return this;
  }

  public int getGroupId() {
    return groupId;
  }

  public LayerTO setGroupId(int groupId) {
    this.groupId = groupId;
    return this;
  }

}
