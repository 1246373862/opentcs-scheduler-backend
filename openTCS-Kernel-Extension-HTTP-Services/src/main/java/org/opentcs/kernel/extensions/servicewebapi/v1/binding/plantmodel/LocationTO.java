/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel;

import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.LinkTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.TripleTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.CoupleTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.PropertyTO;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import static java.util.Objects.requireNonNull;
import javax.annotation.Nonnull;
import org.opentcs.data.model.visualization.LocationRepresentation;

/**
 */
public class LocationTO {

  private String name;
  private String typeName;
  private TripleTO position;
  private List<LinkTO> links = List.of();
  private boolean locked;
  private Layout layout = new Layout();
  private List<PropertyTO> properties = List.of();

  @JsonCreator
  public LocationTO(
      @Nonnull @JsonProperty(value = "name", required = true) String name,
      @Nonnull @JsonProperty(value = "typeName", required = true) String typeName,
      @Nonnull @JsonProperty(value = "position", required = true) TripleTO position
  ) {
    this.name = requireNonNull(name, "name");
    this.typeName = requireNonNull(typeName, "typeName");
    this.position = requireNonNull(position, "position");
  }

  @Nonnull
  public String getName() {
    return name;
  }

  public LocationTO setName(@Nonnull String name) {
    this.name = requireNonNull(name, "name");
    return this;
  }

  @Nonnull
  public List<PropertyTO> getProperties() {
    return properties;
  }

  public LocationTO setProperties(@Nonnull List<PropertyTO> properties) {
    this.properties = requireNonNull(properties, "properties");
    return this;
  }

  @Nonnull
  public String getTypeName() {
    return typeName;
  }

  public LocationTO setTypeName(@Nonnull String typeName) {
    this.typeName = requireNonNull(typeName, "typeName");
    return this;
  }

  @Nonnull
  public TripleTO getPosition() {
    return position;
  }

  public LocationTO setPosition(@Nonnull TripleTO position) {
    this.position = requireNonNull(position, "position");
    return this;
  }

  @Nonnull
  public List<LinkTO> getLinks() {
    return links;
  }

  public LocationTO setLinks(@Nonnull List<LinkTO> links) {
    this.links = requireNonNull(links, "links");
    return this;
  }

  public boolean isLocked() {
    return locked;
  }

  public LocationTO setLocked(boolean locked) {
    this.locked = locked;
    return this;
  }

  @Nonnull
  public Layout getLayout() {
    return layout;
  }

  @Nonnull
  public LocationTO setLayout(@Nonnull Layout layout) {
    this.layout = requireNonNull(layout, "layout");
    return this;
  }

  public static class Layout {

    private CoupleTO position = new CoupleTO(0, 0);
    private CoupleTO labelOffset = new CoupleTO(0, 0);
    private String locationRepresentation = LocationRepresentation.DEFAULT.name();
    private int layerId;

    public Layout() {

    }

    @Nonnull
    public CoupleTO getPosition() {
      return position;
    }

    public Layout setPosition(@Nonnull CoupleTO position) {
      this.position = requireNonNull(position, "position");
      return this;
    }

    @Nonnull
    public CoupleTO getLabelOffset() {
      return labelOffset;
    }

    public Layout setLabelOffset(@Nonnull CoupleTO labelOffset) {
      this.labelOffset = requireNonNull(labelOffset, "labelOffset");
      return this;
    }

    @Nonnull
    public String getLocationRepresentation() {
      return locationRepresentation;
    }

    public Layout setLocationRepresentation(
        @Nonnull String locationRepresentation) {
      this.locationRepresentation = requireNonNull(
          locationRepresentation, "locationRepresentation");
      return this;
    }

    public int getLayerId() {
      return layerId;
    }

    public Layout setLayerId(int layerId) {
      this.layerId = layerId;
      return this;
    }

  }

}
