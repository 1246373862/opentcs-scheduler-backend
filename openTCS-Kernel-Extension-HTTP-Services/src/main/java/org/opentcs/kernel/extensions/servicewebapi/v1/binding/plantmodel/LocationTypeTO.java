/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel;

import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.PropertyTO;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import static java.util.Objects.requireNonNull;
import javax.annotation.Nonnull;
import org.opentcs.data.model.visualization.LocationRepresentation;

/**
 */
public class LocationTypeTO {

  private String name;
  private List<String> allowedOperations = List.of();
  private List<String> allowedPeripheralOperations = List.of();
  private Layout layout = new Layout();
  private List<PropertyTO> properties = List.of();

  @JsonCreator
  public LocationTypeTO(@Nonnull @JsonProperty(value = "name", required = true) String name) {
    this.name = requireNonNull(name, "name");
  }

  @Nonnull
  public String getName() {
    return name;
  }

  public LocationTypeTO setName(@Nonnull String name) {
    this.name = requireNonNull(name, "name");
    return this;
  }

  @Nonnull
  public List<PropertyTO> getProperties() {
    return properties;
  }

  public LocationTypeTO setProperties(@Nonnull List<PropertyTO> properties) {
    this.properties = requireNonNull(properties, "properties");
    return this;
  }

  @Nonnull
  public List<String> getAllowedOperations() {
    return allowedOperations;
  }

  public LocationTypeTO setAllowedOperations(@Nonnull List<String> allowedOperations) {
    this.allowedOperations = requireNonNull(allowedOperations, "allowedOperations");
    return this;
  }

  @Nonnull
  public List<String> getAllowedPeripheralOperations() {
    return allowedPeripheralOperations;
  }

  public LocationTypeTO setAllowedPeripheralOperations(
      @Nonnull List<String> allowedPeripheralOperations) {
    this.allowedPeripheralOperations = requireNonNull(allowedPeripheralOperations,
                                                      "allowedPeripheralOperations");
    return this;
  }

  @Nonnull
  public Layout getLayout() {
    return layout;
  }

  public LocationTypeTO setLayout(@Nonnull Layout layout) {
    this.layout = requireNonNull(layout, "layout");
    return this;
  }

  public static class Layout {

    private String locationRepresentation = LocationRepresentation.NONE.name();

    public Layout() {

    }

    @Nonnull
    public String getLocationRepresentation() {
      return locationRepresentation;
    }

    public Layout setLocationRepresentation(
        @Nonnull String locationRepresentation) {
      this.locationRepresentation = requireNonNull(locationRepresentation,
                                                   "locationRepresentation");
      return this;
    }

  }

}
