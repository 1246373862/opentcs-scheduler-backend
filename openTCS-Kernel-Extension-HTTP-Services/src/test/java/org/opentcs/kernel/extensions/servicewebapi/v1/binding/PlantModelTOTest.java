/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi.v1.binding;

import java.util.List;
import java.util.Set;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.opentcs.kernel.extensions.servicewebapi.JsonBinder;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.BlockTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.CoupleTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.LayerGroupTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.LayerTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.LocationTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.LocationTypeTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.PathTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.PeripheralOperationTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.PointTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.EnvelopeTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.PropertyTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.shared.TripleTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.VehicleTO;
import org.opentcs.kernel.extensions.servicewebapi.v1.binding.plantmodel.VisualLayoutTO;

/**
 */
public class PlantModelTOTest {

  private JsonBinder jsonBinder;

  @BeforeEach
  public void setUp() {
    jsonBinder = new JsonBinder();
  }

  @Test
  public void jsonSample() {
    PlantModelTO to = new PlantModelTO("some-name")
        .setPoints(List.of(
            new PointTO("some-point")
                .setProperties(List.of(new PropertyTO("point-prop", "point-value")))
                .setPosition(new TripleTO(25000, -15000, 0))
                .setVehicleOrientationAngle(73.3)
                .setType("PARK_POSITION")
                .setVehicleEnvelopes(
                    List.of(
                        new EnvelopeTO(
                            "envelopeType-01",
                            List.of(
                                new CoupleTO(25500, -15500),
                                new CoupleTO(25500, -14500),
                                new CoupleTO(24500, -14500),
                                new CoupleTO(24500, -15500)
                            )
                        )
                    )
                )
                .setLayout(new PointTO.Layout()
                    .setPosition(new CoupleTO(25000, -15000))
                    .setLabelOffset(new CoupleTO(-10, -20))
                    .setLayerId(0)),
            new PointTO("some-point2")
                .setProperties(List.of(new PropertyTO("point-prop", "point-value")))
                .setPosition(new TripleTO(18000, -15000, 0))
                .setLayout(new PointTO.Layout()
                    .setPosition(new CoupleTO(18000, -15000))
                    .setLabelOffset(new CoupleTO(-10, -20))
                    .setLayerId(0)),
            new PointTO("some-point3")
                .setProperties(List.of(new PropertyTO("point-prop", "point-value")))
                .setPosition(new TripleTO(25000, -9000, 0))
                .setLayout(new PointTO.Layout()
                    .setPosition(new CoupleTO(25000, -9000))
                    .setLabelOffset(new CoupleTO(-10, -20))
                    .setLayerId(0))))
        .setPaths(List.of(
            new PathTO("some-path", "some-point", "some-point2")
                .setProperties(List.of(new PropertyTO("path-prop", "path-value")))
                .setLength(3)
                .setMaxVelocity(13)
                .setMaxReverseVelocity(3)
                .setLocked(true)
                .setVehicleEnvelopes(
                    List.of(
                        new EnvelopeTO(
                            "envelopeType-01",
                            List.of(
                                new CoupleTO(25500, -15500),
                                new CoupleTO(25500, -14500),
                                new CoupleTO(17500, -14500),
                                new CoupleTO(17500, -15500)
                            )
                        )
                    )
                )
                .setPeripheralOperations(List.of(new PeripheralOperationTO("some-op",
                                                                           "some-location")
                    .setExecutionTrigger("AFTER_ALLOCATION")
                    .setCompletionRequired(true)))
                .setLayout(new PathTO.Layout()
                    .setConnectionType("SLANTED")
                    .setLayerId(0)
                    .setControlPoints(List.of(new CoupleTO(43000, 30000),
                                              new CoupleTO(44000, 31000),
                                              new CoupleTO(45000, 32000)))),
            new PathTO("another-path", "some-point2", "some-point3")))
        .setLocationTypes(List.of(
            new LocationTypeTO("some-locationType")
                .setProperties(List.of(new PropertyTO("locType-prop", "locType-value")))
                .setAllowedOperations(List.of("some-operation", "another-operation", "operation3"))
                .setAllowedPeripheralOperations(List.of("some-perOp", "another-perOp", "perOp3"))
                .setLayout(
                    new LocationTypeTO.Layout()
                        .setLocationRepresentation("WORKING_GENERIC"))))
        .setLocations(List.of(
            new LocationTO("some-location",
                           "some-locationType",
                           new TripleTO(30000, -15000, 0))
                .setLocked(true)
                .setLayout(
                    new LocationTO.Layout()
                        .setPosition(new CoupleTO(30000, -15000))
                        .setLabelOffset(new CoupleTO(-10, -20))
                        .setLocationRepresentation("LOAD_TRANSFER_GENERIC"))))
        .setBlocks(List.of(
            new BlockTO("some-block")
                .setProperties(List.of(new PropertyTO("block-prop", "block-value")))
                .setType("SAME_DIRECTION_ONLY")
                .setMemberNames(Set.of("some-point2"))
                .setLayout(new BlockTO.Layout())))
        .setVehicles(List.of(
            new VehicleTO("some-vehicle")
                .setProperties(List.of(new PropertyTO("vehicle-prop", "vehicle-value")))
                .setLength(1456)
                .setEnergyLevelCritical(10)
                .setEnergyLevelGood(30)
                .setEnergyLevelSufficientlyRecharged(60)
                .setMaxVelocity(2000)
                .setMaxReverseVelocity(733)
                .setLayout(new VehicleTO.Layout().setRouteColor("#123456"))))
        .setVisualLayout(
            new VisualLayoutTO("some-visualLayout")
                .setProperties(List.of(new PropertyTO("vLayout-prop", "vLayout-value")))
                .setScaleX(65)
                .setScaleY(65)
                .setLayers(List.of(new LayerTO(0, 0, true, "layer0", 0)))
                .setLayerGroups(List.of(new LayerGroupTO(0, "layerGroup0", true))))
        .setProperties(List.of(new PropertyTO("plantModel-prop", "value")));

    Approvals.verify(jsonBinder.toJson(to));
  }

}
