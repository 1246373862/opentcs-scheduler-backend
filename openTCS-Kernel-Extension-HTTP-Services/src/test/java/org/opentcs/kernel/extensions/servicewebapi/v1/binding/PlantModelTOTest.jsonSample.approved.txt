{
  "name" : "some-name",
  "points" : [ {
    "name" : "some-point",
    "position" : {
      "x" : 25000,
      "y" : -15000,
      "z" : 0
    },
    "vehicleOrientationAngle" : 73.3,
    "type" : "PARK_POSITION",
    "layout" : {
      "position" : {
        "x" : 25000,
        "y" : -15000
      },
      "labelOffset" : {
        "x" : -10,
        "y" : -20
      },
      "layerId" : 0
    },
    "vehicleEnvelopes" : [ {
      "key" : "envelopeType-01",
      "vertices" : [ {
        "x" : 25500,
        "y" : -15500
      }, {
        "x" : 25500,
        "y" : -14500
      }, {
        "x" : 24500,
        "y" : -14500
      }, {
        "x" : 24500,
        "y" : -15500
      } ]
    } ],
    "properties" : [ {
      "name" : "point-prop",
      "value" : "point-value"
    } ]
  }, {
    "name" : "some-point2",
    "position" : {
      "x" : 18000,
      "y" : -15000,
      "z" : 0
    },
    "vehicleOrientationAngle" : "NaN",
    "type" : "HALT_POSITION",
    "layout" : {
      "position" : {
        "x" : 18000,
        "y" : -15000
      },
      "labelOffset" : {
        "x" : -10,
        "y" : -20
      },
      "layerId" : 0
    },
    "vehicleEnvelopes" : [ ],
    "properties" : [ {
      "name" : "point-prop",
      "value" : "point-value"
    } ]
  }, {
    "name" : "some-point3",
    "position" : {
      "x" : 25000,
      "y" : -9000,
      "z" : 0
    },
    "vehicleOrientationAngle" : "NaN",
    "type" : "HALT_POSITION",
    "layout" : {
      "position" : {
        "x" : 25000,
        "y" : -9000
      },
      "labelOffset" : {
        "x" : -10,
        "y" : -20
      },
      "layerId" : 0
    },
    "vehicleEnvelopes" : [ ],
    "properties" : [ {
      "name" : "point-prop",
      "value" : "point-value"
    } ]
  } ],
  "paths" : [ {
    "name" : "some-path",
    "srcPointName" : "some-point",
    "destPointName" : "some-point2",
    "length" : 3,
    "maxVelocity" : 13,
    "maxReverseVelocity" : 3,
    "peripheralOperations" : [ {
      "operation" : "some-op",
      "locationName" : "some-location",
      "executionTrigger" : "AFTER_ALLOCATION",
      "completionRequired" : true
    } ],
    "locked" : true,
    "layout" : {
      "connectionType" : "SLANTED",
      "controlPoints" : [ {
        "x" : 43000,
        "y" : 30000
      }, {
        "x" : 44000,
        "y" : 31000
      }, {
        "x" : 45000,
        "y" : 32000
      } ],
      "layerId" : 0
    },
    "vehicleEnvelopes" : [ {
      "key" : "envelopeType-01",
      "vertices" : [ {
        "x" : 25500,
        "y" : -15500
      }, {
        "x" : 25500,
        "y" : -14500
      }, {
        "x" : 17500,
        "y" : -14500
      }, {
        "x" : 17500,
        "y" : -15500
      } ]
    } ],
    "properties" : [ {
      "name" : "path-prop",
      "value" : "path-value"
    } ]
  }, {
    "name" : "another-path",
    "srcPointName" : "some-point2",
    "destPointName" : "some-point3",
    "length" : 1,
    "maxVelocity" : 0,
    "maxReverseVelocity" : 0,
    "peripheralOperations" : [ ],
    "locked" : false,
    "layout" : {
      "connectionType" : "DIRECT",
      "controlPoints" : [ ],
      "layerId" : 0
    },
    "vehicleEnvelopes" : [ ],
    "properties" : [ ]
  } ],
  "locationTypes" : [ {
    "name" : "some-locationType",
    "allowedOperations" : [ "some-operation", "another-operation", "operation3" ],
    "allowedPeripheralOperations" : [ "some-perOp", "another-perOp", "perOp3" ],
    "layout" : {
      "locationRepresentation" : "WORKING_GENERIC"
    },
    "properties" : [ {
      "name" : "locType-prop",
      "value" : "locType-value"
    } ]
  } ],
  "locations" : [ {
    "name" : "some-location",
    "typeName" : "some-locationType",
    "position" : {
      "x" : 30000,
      "y" : -15000,
      "z" : 0
    },
    "links" : [ ],
    "locked" : true,
    "layout" : {
      "position" : {
        "x" : 30000,
        "y" : -15000
      },
      "labelOffset" : {
        "x" : -10,
        "y" : -20
      },
      "locationRepresentation" : "LOAD_TRANSFER_GENERIC",
      "layerId" : 0
    },
    "properties" : [ ]
  } ],
  "blocks" : [ {
    "name" : "some-block",
    "type" : "SAME_DIRECTION_ONLY",
    "layout" : {
      "color" : "#FF0000"
    },
    "memberNames" : [ "some-point2" ],
    "properties" : [ {
      "name" : "block-prop",
      "value" : "block-value"
    } ]
  } ],
  "vehicles" : [ {
    "name" : "some-vehicle",
    "length" : 1456,
    "energyLevelCritical" : 10,
    "energyLevelGood" : 30,
    "energyLevelFullyRecharged" : 90,
    "energyLevelSufficientlyRecharged" : 60,
    "maxVelocity" : 2000,
    "maxReverseVelocity" : 733,
    "layout" : {
      "routeColor" : "#123456"
    },
    "properties" : [ {
      "name" : "vehicle-prop",
      "value" : "vehicle-value"
    } ]
  } ],
  "visualLayout" : {
    "name" : "some-visualLayout",
    "scaleX" : 65.0,
    "scaleY" : 65.0,
    "layers" : [ {
      "id" : 0,
      "ordinal" : 0,
      "visible" : true,
      "name" : "layer0",
      "groupId" : 0
    } ],
    "layerGroups" : [ {
      "id" : 0,
      "name" : "layerGroup0",
      "visible" : true
    } ],
    "properties" : [ {
      "name" : "vLayout-prop",
      "value" : "vLayout-value"
    } ]
  },
  "properties" : [ {
    "name" : "plantModel-prop",
    "value" : "value"
  } ]
}
